src/Collection.php

```diff
@@ -0,0 +1,90 @@
+<?php namespace TightenCo\Jigsaw;
+
+use Illuminate\Support\Collection as BaseCollection;
+
+class Collection extends BaseCollection
+{
+    private $settings;
+    public $name;
+
+    public static function withSettings($settings, $name)
+    {
+        $collection = new static;
+        $collection->settings = $settings;
+        $collection->name = $name;
+
+        return $collection;
+    }
+
+    public function loadItems($items)
+    {
+        $sortedItems = $this->defaultSort($items)->keyBy(function($item) {
+            return $item->filename;
+        });
+        parent::__construct($this->addAdjacentItems($sortedItems));
+
+        return $this;
```

Comment:
Since it's an Illuminate Collection, the call to the constructor is hydrating the collection with its children `CollectionItem`s (i.e. calling `$this->items = $this->getArrayableItems($items);`).

The `CollectionDataLoader` instantiates a new `Collection` using the named constructor `withSettings`, storing collection-specific settings, but not storing any items yet. The Loader then builds each `CollectionItem`—giving each of them a reference to its parent `Collection`—and then loads the items into the `Collection` with `loadItems` method.

An alternative would be to instantiate the `Collection` and load the items in the same call, which would then require looping over the items to store a reference to the parent `Collection` on each one.